### ISSUE TITLE
Conflicting line length rule with scalafmt

### ISSUE THREAD

**Issue by arjun-1** [liancheng/scalafix-organize-imports#43]
So far this plugin works really nice :slightly_smiling_face: . I only encountered the following issue. I have an import statement as follows:
```
import sttp.client.{
  DeserializationError,
  HttpError,
  NothingT,
  Response,
  ResponseAs,
  ResponseError,
  SttpBackend,
  basicRequest
}
```
scalafix-organize-imports tries to rewrite it to 
```
import sttp.client.{DeserializationError, HttpError, NothingT, Response, ResponseAs, ResponseError, SttpBackend, basicRequest}
```
while scalafmt insists on the original formatting. My `.scalafmt.conf`:
```
version = "2.4.1"

align = more
maxColumn = 100
rewrite {
  rules = [
    SortModifiers,
    RedundantBraces,
    PreferCurlyFors
  ]

  redundantBraces.stringInterpolation = true
}
```
and `.scalafix.conf`:
```
rules = [
  NoAutoTupling,
  DisableSyntax,
  LeakingImplicitClassVal,
  NoValInForComprehension,
  ProcedureSyntax,
  OrganizeImports
]

OrganizeImports {
  groupedImports = Merge
}
```

**Comment by liancheng** [comment]
Hey @arjun-1, thanks for trying it out and reporting!

This is a known limitation of the `OrganizeImports` rule. For rules making surgical changes, e.g., the Scalafix built-in `NoAutoTupling` and `ProcedureSyntax` rules, it's relatively easy to respect the original formatting convention. However, `OrganizeImports` shuffles a lot more text, and it's harder to meet that requirement.

Basically, `OrganizeImports` generates two big Scalafix patches to do the job:

1. One patch for removing the original global imports.
2. Another patch for inserting the organized imports as a big trunk of string to the original location.

   This patch is generated by pretty-printing transformed AST nodes, so it's pretty hard to keep the original format.

One straightforward workflow to avoid this issue is to run Scalafix with `OrganizeImports` first and then run Scalafmt to reformat your code base. The former performs semantic rewriting while the latter performs syntactic rewriting. Let's just have the right tool to do the right job!

**Comment by arjun-1** [comment]
Thanks for the clarification. We enforce linting in our CI pipelines (using both `scalafix --check` and `scalafmtCheckAll`). I guess the best way to go about this would be to to exclude `OrganizeImports` from the scalafix rules list, while maybe adding `;scalafix OrganizeImports; test:scalafix OrganizeImports;` to the linting command alias

**Comment by liancheng** [comment]
@arjun-1, although right now it seems to be obvious, I didn't realize that you were talking about linting, which is a legitimate use case `OrganizeImports` should support well.

In the scope of linting, it is much easier to implement, since what we need to do is simply leaving an import untouched if it's already organized.

**Comment by liancheng** [comment]
PR #44 should fix this issue.

**Comment by arjun-1** [comment]
Hi @liancheng , thanks for the quick reply  :slightly_smiling_face: . I tried your fix with the snapshot version, but strangely enough it did not make a difference with 0.3.0. (Given that it seems to work for the given input/output of https://github.com/liancheng/scalafix-organize-imports/pull/44 , I really wonder whether something went wrong on my side of the configuration :thinking: ).

In any case I made a minimal example which reproduces the issue:
https://github.com/arjun-1/failing-scalafix-organize-imports-lint
Just run `scalafmtAll` and `scalafix`, and the imports will be rewritten to a single line.

**Comment by liancheng** [comment]
@arjun-1, thank you so much for the minimal reproduction! I figured out why it didn't work for you. It's because you are using Scala 2.13 while `OrganizeImports` is built using Scala 2.12.

`OrganizeImports` calls `importer.syntax` to pretty-print an import expression. Internally, the `syntax` method accepts an implicit `scala.meta.Dialect` parameter. While pretty-printing, the `syntax` method checks two things:

1. Whether this AST node originates directly from the parser.
2. Whether the Scala dialect of the original source being parsed comforms to the implicit dialect.

When both conditions are true, it simply returns the original source text being parsed. Otherwise, it pretty-prints the AST node using the implicit dialect.

`OrganizeImports` is built using Scala 2.12, so at runtime, this implicit dialect is by default Scala 2.12. Therefore, condition 2 doesn't hold.

Instead, we can simply replace the `importer.syntax` with `importer.toString`. The latter doesn't check condition 2. Instead, it always uses Scala 2.11 dialect, which is good enough for pretty-printing import statements.

**Comment by liancheng** [comment]
@arjun-1, let's see whether the snapshot release `0.3.0+5-98c374f1-SNAPSHOT` fixes your issue.

**Comment by arjun-1** [comment]
Excellent! Indeed the issue is resolved with 0.3.0+5-98c374f1-SNAPSHOT :smiley:
